### ISSUE TITLE
Feature request: list-like behaviour for `Sequence`

### ISSUE THREAD

**Issue by davidgilbertson** [huggingface/tokenizers#1175]
When using a tokenizer like `ByteLevelBPETokenizer` it might be initialised with a `Sequence` of normalizers, and I may want to add to this list. It would be neat if `Sequence` behaved [like `ModuleList` in PyTorch](https://github.com/pytorch/pytorch/blob/3623cfb79095b1a3b6b41b9a5edea278d0a619ef/torch/nn/modules/container.py#L250), with it's own list-like behaviour, so I could do:
```py
tokenizer.normalizer.append(SomeNewThing())
```
Even better would be if it behaved like a [scikit-learn `Pipeline`](https://github.com/scikit-learn/scikit-learn/blob/7e8d3c505baafa45bd94d9e28ee3d35141ff91c4/sklearn/pipeline.py#L279) that allows partial application like `pipeline[:-1]`

Even if there was just some way to get at the contents of a normalizer sequence, I could then do this:
```py
tokenizer.normalizer = normalizers.Sequence(
    [
        *tokenizer.normalizer.items(),
        *more_stuff,
    ]
)
```

**Comment by Narsil** [comment]
Thanks for the suggestion. It's a great suggestion, but I'm don't really have the time to implement such things at the moment.

Cheers !

**Comment by SandeepaDevin** [comment]
I can look into this and give it a try, if I can get an idea of where to look at.

**Comment by Narsil** [comment]
@SandeepaDevin Thanks for proposing help ! 

The file you want to look at is here: https://github.com/huggingface/tokenizers/blob/main/bindings/python/src/normalizers.rs
This is just for normalizers as asked in the issue but probably all our objects are affected.

There's several layers of things we could do.

I would start small and just having a nice `append` on the Sequence like object would be the easiest I think.

Bonus points if you manage to make it little code, and making sure all our sequences are modified in such a way. :)

**Comment by bharathk005** [comment]
@Narsil 
Is this already implemented here:
https://github.com/huggingface/tokenizers/blob/150559b61e5a9db86724b02d79272b1f7c3ff1af/bindings/python/src/normalizers.rs#L345
